File: backend/src/main/java/com/nearchrist/backend/dto/UserDto.java
package com.nearchrist.backend.dto;

import java.util.Set;

public record UserDto(
        Long id,
        String userName,  // The actual username for display
        String userEmail, // The email, used for login and communication
        String firstName,
        String lastName,
        boolean enabled,
        Set<String> roles
) {}

File: frontend/src/app/user-maintenance/user.dto.ts
export interface UserDto {
  id: number;
  userFullName: string;
  userEmail: string;
  roles: string[];
  enabled: boolean;
}

export interface UserUpsertDto {
  userFullName: string;
  userEmail:string;
  password?: string;
  roles: string[];
}
